include/pcp/common/axis_aligned_bounding_box.hpp

```diff
@@ -135,7 +169,7 @@ kd_bounding_box(ForwardIter begin, ForwardIter end, CoordinateMap const& coordin
     for (auto i = 0u; i < aabb.min.size(); ++i)
     {
         aabb.min[i] = std::numeric_limits<scalar_type>::max();
-        aabb.max[i] = std::numeric_limits<scalar_type>::min();
+        aabb.max[i] = std::numeric_limits<scalar_type>::lowest();
     }
 
     std::array<std::mutex, K> min_mutex;
```

Comment:
Use shared_mutex here. Use std::shared_lock when reading the aabb.min or aabb.max and use std::unique_lock when writing to aabb.min and aabb.max. See [https://en.cppreference.com/w/cpp/thread/shared_mutex](https://en.cppreference.com/w/cpp/thread/shared_mutex)